### PR TITLE
docs: Add JOSS paper

### PR DESCRIPTION
**Pull request by rkansal47**
# Description

Add `paper.md` and `paper.bib` as needed for a [JOSS paper submission](https://joss.readthedocs.io/en/latest/submitting.html). 
Also adds the [Open Journals Github Action](https://joss.readthedocs.io/en/latest/submitting.html#github-action) to compile the paper.

# Action Item Checklist

 - [x] Write paper
 - [x] Check it compiles
 - [x] Review
 - [x] Submit to JOSS

# Paper checklist

 - [x] A list of the authors of the software and their affiliations, using the correct format.
 - [x] A summary describing the high-level functionality and purpose of the software for a diverse, non-specialist audience.
 - [x] A clear Statement of Need that illustrates the research purpose of the software.
 - [x] A list of key references, including to other software addressing related needs.
 - [x] Mention (if applicable) a representative set of past or ongoing research projects using the software and recent scholarly publications enabled by it.
 - [x]  Acknowledgement of any financial support.

### CODE REVIEW

**Comment by jmduarte** [comment]
We should also add the NeurIPS paper to CITATION.cff as a "reference" I guess.

File: .github/workflows/JOSS-pdf.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    name: JOSS Draft
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build draft JOSS PDF
+        uses: openjournals/openjournals-draft-action@master
+        with:
+          journal: joss
+          # This should be the path to the paper within your repo.
+          paper-path: paper.md
```
Comment (jmduarte) [inline .github/workflows/JOSS-pdf.yml:18]: not that it matters that much anymore, but shouldn't this be `docs/JOSS/paper.md` ?

File: .github/workflows/JOSS-pdf.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Upload
+        uses: actions/upload-artifact@v1
+        with:
+          name: paper
+          # This is the output path where Pandoc will write the compiled
+          # PDF. Note, this should be the same directory as the input
+          # paper.md
+          path: paper.pdf
```
Comment (jmduarte) [inline .github/workflows/JOSS-pdf.yml:26]: Ditto: `docs/JOSS/paper.pdf`

**Review by jmduarte — CHANGES_REQUESTED** [review]

File: .github/workflows/JOSS-pdf.yml (hunk 1 above)
Comment (rkansal47) [reply .github/workflows/JOSS-pdf.yml]: Yup, thanks

**Review by jmduarte — APPROVED** [review]

**Pull request merged**